packages/core/src/browser/preferences/preference-contribution.ts

```diff
@@ -74,21 +79,25 @@ export class PreferenceSchemaProvider extends PreferenceProvider {
         this.preferenceContributions.getContributions().forEach(contrib => {
             this.doSetSchema(contrib.schema);
         });
+        this.combinedSchema.additionalProperties = false;
         this.updateValidate();
         this._ready.resolve();
     }
 
     protected doSetSchema(schema: PreferenceSchema): void {
         const props: string[] = [];
-        for (const property in schema.properties) {
+        for (const property of Object.keys(schema.properties)) {
```

Comment:
`in` operator is an equivalent of `of Object.keys`, just shorter

Reply:
i got an eslint error `for (... in ...) statements must be filtered with an if statement (forin)`  when `for...in` was used.